src/lib/containers/table/SynapseTable.tsx

```diff
@@ -182,9 +182,9 @@ export default class SynapseTable extends React.Component<
     }
     return shouldComponentUpdate
   }
-  componentDidUpdate(prevProps: QueryWrapperChildProps & SynapseTableProps) {
+  componentDidUpdate(prevProps: QueryWrapperChildProps & SynapseTableProps, prevState: Readonly<SynapseTableState>) {
     // PORTALS-2081: if the data changed, then get the new entity headers
-    if (!eq(prevProps.data, this.props.data)) {
+    if (!eq(prevProps.data, this.props.data) || !eq(prevState.isEntityView, this.state.isEntityView)) {
```

Comment:
Technically, I think this would be !XOR, so !(prevState.isEntityView ^ this.state.isEntityView), but I think using lodash eq is more readable.

Reply:
There are so many questions that come to mind about how we could probably overhaul this so it's easy to follow. It seems weird to me that we have to check both the data and the type, but I'm guessing it has to do with the fact that isEntityView is determined in state asynchronously.

So this works for now!

Reply:
That's exactly right, it's because isEntityView is stored in state and determined asynchronously